homeassistant/components/rotel/media_player.py

```diff
@@ -0,0 +1,493 @@
+                'HDMI', 'Coaxial', 'Optical', 'A', '1', '2', '3', '4', '5')}
+        # Not actually sure what these are.
+        # Might move them if I ever work it out.
+        self._misc_icons = {
+            k: binary_sensor_value(message.icons[k]) for k in ('<', '>')}
+        self.async_schedule_update_ha_state()
+
+    def handle_trigger_message(self, message):
+        """Map trigger message to object attributes."""
+        self._triggers = message.flags_to_list(message.flags)
+        self.async_schedule_update_ha_state()
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def supported_features(self):
+        """Flag media player features that are supported."""
+        return SUPPORT_ROTEL_RSP1570
+
+    @property
+    def assumed_state(self):
+        """Indicate that state is assumed."""
+        return True
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    async def async_turn_on(self):
+        """Turn the media player on."""
+        await self._conn.send_command('POWER_ON')
+        self._state = STATE_ON
+
+    async def async_turn_off(self):
+        """Turn off media player."""
+        await self._conn.send_command('POWER_OFF')
+        self._state = STATE_OFF
+
+    @property
+    def source_list(self):
+        """Return the list of available input sources."""
+        return sorted((self._sources_to_select.keys()))
+
+    @property
+    def source(self):
+        """Return the current input source."""
+        return self._source_name
+
+    async def async_select_source(self, source):
+        """Select input source."""
+        await self._conn.send_command(self._sources_to_select[source])
+
+    async def async_volume_up(self):
+        """Volume up media player."""
+        await self._conn.send_command('VOLUME_UP')
+
+    async def async_volume_down(self):
+        """Volume down media player."""
+        await self._conn.send_command('VOLUME_DOWN')
+
+    async def async_mute_volume(self, mute):
+        """Mute (true) or unmute (false) media player."""
+        # Note that the main zone has a MUTE_ON and MUTE_OFF command
+        # Could switch to that instead but for now sticking with the
+        # regular mute commands that affect whatever zone is on the
+        # info display
+        if self._mute_on is None:
+            # Chances are that this is the right thing to do
+            await self._conn.send_command('MUTE_TOGGLE')
+        elif self._mute_on and not mute:
+            await self._conn.send_command('MUTE_TOGGLE')
+        elif not self._mute_on and mute:
+            await self._conn.send_command('MUTE_TOGGLE')
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific state attributes."""
+        if self.state == STATE_OFF:
+            return None
+        attributes = {
```

Comment:
Do not duplicate items here that are already represented -- such as source name and volume by the media player component.  Also the icon lists are not appropriate (that's a front-end concern) and should be removed as that is not a state of the device.

Reply:
A quick question about this one too.  Why do you say that the icons aren't a state of the device?   They reflect the front panel icons on the device that dynamically indicate what the device is doing at any point in time.   Things like sound mode, which speakers are active etc.    Not all of them feed into the current media control card but I was hoping to use them to create a custom card that looked as much like the front panel as I could manage.   In the meantime, in my current setup I have some of them set up as template sensors.

One could argue that some of the icons could be used to infer the sound_mode attribute of the MediaPlayer object but unfortunately they aren't sufficiently detailed to do that so I felt that it was better to just reflect what is available.

Would it be more acceptable if I only created these states if there was a config entry to enable them?

Reply:
Are there no replacement indicators of device state instead of the icons. It seems weird to use icons to represent state in the backend.